app/code/Magento/Developer/Model/TemplateEngine/Plugin/DebugHints.php

```diff
@@ -58,24 +59,51 @@ class DebugHints
     protected $debugHintsPath;
 
     /**
+     * XPath of configuration of the debug hints show with parameter
+     *
+     *     dev/debug/template_hints_storefront_show_with_parameter
+     *
+     * @var string
+     */
+    protected $debugHintsShowWithParameter;
+
+    /**
+     * XPath of configuration of the debug hints URL parameter
+     *
+     *     dev/debug/template_hints_parameter_value
+     *
+     * @var string
+     */
+    protected $debugHintsParameter;
+
+    /**
      * @param ScopeConfigInterface $scopeConfig
      * @param StoreManagerInterface $storeManager
      * @param DevHelper $devHelper
      * @param DebugHintsFactory $debugHintsFactory
      * @param string $debugHintsPath
+     * @param Http $http
+     * @param string $debugHintsShowWithParameter
+     * @param string $debugHintsParameter
      */
     public function __construct(
         ScopeConfigInterface $scopeConfig,
         StoreManagerInterface $storeManager,
         DevHelper $devHelper,
         DebugHintsFactory $debugHintsFactory,
-        $debugHintsPath
+        $debugHintsPath,
+        Http $http,
```

Comment:
Yes, sure. There's an example in the guides:
http://devdocs.magento.com/guides/v2.0/contributor-guide/backward-compatible-development/#ui-id-3